homeassistant/components/generic/config_flow.py

```diff
@@ -168,10 +168,15 @@ async def async_test_still(hass, info) -> tuple[dict[str, str], str | None]:
     return {}, f"image/{fmt}"
 
 
-def slug_url(url) -> str | None:
+def slug_url(hass, url) -> str | None:
     """Convert a camera url into a string suitable for a camera name."""
     if not url:
         return None
+    if not isinstance(url, template_helper.Template):
+        url = cv.template(url)
+        url.hass = hass
+    # We shouldn't have any exceptions here, because we verified the url earlier
```

Comment:
A template that renders a valid URL can be an invalid URL itself.

```jinja2
{% if x %}https://bla{% else %}https://yo{% end %}
```

Reply:
I mean that by this point in the config flow we should have already rendered the url and downloaded something from it once during async_test_still  But in reality thats only for still at the moment.  Probably should handle the possible exceptions  here.

Reply:
Done.  Test added for this [b1ca479](https://github.com/home-assistant/core/pull/71031/commits/b1ca47953e64f5b2afbc08cf310d1954dafaf7dd)